### DISCUSSION TITLE
question about checkDataValidity

### DISCUSSION THREAD

**Discussion by thevelvetunderground** [gimli-org/pyGIMLi#549] [Q&A]
I follow the tutorials like this:

data.checkDataValidity()    
print(data["valid"])
data.removeInvalid()        
print(data)
print("Rx:",data["Rx"],"Tx:",data["Tx"])          
sx = pg.x(data)    
data["dist"] = np.abs(sx[data["Rx"]] - sx[data["Tx"]])
print(data["dist"])

But it seems clear all the data index:
<img width="488" alt="image" src="https://github.com/gimli-org/gimli/assets/91045087/425514d9-d0f8-4847-b355-b2d95015b353">

Is this a bug or I overlookde some code?

**Comment by carsten-forty2** [comment]
It shows no data so it can't calculate distances. Could you provide the data so we can check why they had been all removed?

**Reply by thevelvetunderground** [reply]
Hi, I followed the tutorials see:https://www.pygimli.org/_tutorials_auto/0_basics/plot_3-data_container.html. And the complete code is as follows:
```
import numpy as np
import matplotlib.pyplot as plt
import pygimli as pg
from pygimli.physics import VESManager
data = pg.DataContainer()
data.registerSensorIndex("Tx")
data.registerSensorIndex("Rx")
for x in np.arange(10):
    data.createSensor([x*2, 0]) 
data["Rx"] = np.arange(12)
data["Tx"] = pg.Vector(data.size(), 2)
print(data)
data.save("TxRx.data")
print(open("TxRx.data").read())
data.checkDataValidity()    
print(data["valid"])
data.removeInvalid()       
print("Rx:",data["Rx"],"Tx:",data["Tx"])         
sx = pg.x(data)    
data["dist"] = np.abs(sx[data["Rx"]] - sx[data["Tx"]])
print(data["dist"])
```
seems like data.checkDataValidity()  and data.removeInvalid()   will clear all the data

**Comment by halbmy** [comment]
As the code clearly shows, by default the data row `data["valid"]` is zero and the method `data.checkDataValidity()` does not check this, instead it keeps this and sets additional data invalid. This behaviour is of course counter-intuitive and should be corrected.

For now you should just set the data valid by `data["valid"]=1` (or `data["valid"]=pg.Vector(data.size(), 1)`).
